### PR TITLE
feat(ihev1): turn off ihev1 outbound

### PR DESCRIPTION
**Pull request by jonahkaye**
Refs: #[1667](https://github.com/metriport/metriport-internal/issues/1667)

### Description

- deprecate ihev1
- Deprecate FFs too

### Testing

- Staging
  - [ ] _testing step 1_

### Release Plan

- [ ] Merge this
- [ ] Remove FFs in production, staging, sandbox for cxsWithIheV2Enabled and oidsWithIheV2Enabled only after PR has been shipped to production

### CODE REVIEW

File: packages/api/src/external/carequality/document/process-outbound-document-query-resps.ts [hunk 1]
```diff
@@ -155,46 +148,24 @@ export async function processOutboundDocumentQueryResps({
...
+    const outboundDocumentQueryResults: OutboundDocumentQueryResp[] = [];
 
     for (const result of resultsWithMetriportIdAndDrUrl) {
-      if (isV2EnabledForCx || v2GatewayOIDs.includes(result.gateway.homeCommunityId)) {
-        outboundDocumentQueryResultsV2.push(result);
-      } else {
-        outboundDocumentQueryResultsV1.push(result);
-      }
+      outboundDocumentQueryResults.push(result);
     }
```
Comment (leite08) [inline packages/api/src/external/carequality/document/process-outbound-document-query-resps.ts:155]: We can remove this loop altogether, right?

File: packages/api/src/external/carequality/patient.ts [hunk 2]
```diff
@@ -175,7 +144,6 @@ async function prepareForPatientDiscovery(
   });
 
   return {
-    pdRequestGatewayV1,
     pdRequestGatewayV2,
   };
```
Comment (leite08) [inline packages/api/src/external/carequality/patient.ts:148]: Now could return a single array, right?

File: packages/api/src/external/carequality/gateway/index.ts [hunk 3]
```diff
@@ -50,10 +49,9 @@ export async function gatherXCPDGateways(patient: Patient): Promise<Gateways> {
   const filteredOrgs = facilitiesWithEpicFilter(allOrgs, isEpicEnabled);
   const allOrgsWithBasics = filteredOrgs.map(toBasicOrgAttributes);
   const orgsToSearch = filterCQOrgsToSearch(allOrgsWithBasics);
-  const { v1Gateways, v2Gateways } = await cqOrgsToXCPDGateways(orgsToSearch, patient.cxId);
+  const { v2Gateways } = await cqOrgsToXCPDGateways(orgsToSearch);
 
   return {
-    v1Gateways,
     v2Gateways,
   };
```
Comment (leite08) [inline packages/api/src/external/carequality/gateway/index.ts:56]: ditto: could return a single array now?

File: packages/api/src/external/carequality/organization-conversion.ts [hunk 4]
```diff
@@ -1,35 +1,22 @@
...
-      } else {
-        v1Gateways.push(gateway);
-      }
+      v2Gateways.push(gateway);
     }
   }
   return {
-    v1Gateways,
     v2Gateways,
   };
```
Comment (leite08) [inline packages/api/src/external/carequality/organization-conversion.ts:21]: ditto: single array?

File: packages/api/src/external/carequality/shared.ts [hunk 5]
```diff
@@ -19,39 +17,33 @@ export function isGWValid(gateway: { homeCommunityId: string; url: string }): bo
   return !!gateway.homeCommunityId && !!gateway.url;
 }
 
-export async function validateCQEnabledAndInitGW(
+export async function validateCQEnabled(
```
Comment (leite08) [inline packages/api/src/external/carequality/shared.ts:20]: Maybe rename to `isCqEnabled`?

**Review by leite08 — COMMENTED** [review]
### re: staged release

Did we already move all GWs to v2 and this is just removing the code?

### re: release plan

Great that you indicated the name of the FFs to be removed!

But it has removing FFs before merging the PR - it can only be removed _after_ its released, right? I recommend we're very specific w/ the release plan so (1) we don't need to decide things when we're shipping stuff and (2) others can ship it for us.

In this case, maybe we only remove the FFs an hour or so after the release is complete?

**Comment by jonahkaye** [comment]
> ### re: staged release
> Did we already move all GWs to v2 and this is just removing the code?

Yessir!

> ### re: release plan
> Great that you indicated the name of the FFs to be removed!
> 
> But it has removing FFs before merging the PR - it can only be removed _after_ its released, right? I recommend we're very specific w/ the release plan so (1) we don't need to decide things when we're shipping stuff and (2) others can ship it for us.
> 
> In this case, maybe we only remove the FFs an hour or so after the release is complete?

Will do with this update!

**Review by leite08 — APPROVED** [review]

**Pull request merged**